int32       key
int32       gen_hash
float64     score
bool        at_light
int32[]     state_sequence
geometry_msgs/Point[] locations